test/compiler/compiler.jl

```diff
@@ -2086,3 +2086,13 @@ let ci = code_typed(bar_inlining_apply, Tuple{})[1].first
     @test length(ci.code) == 2
     @test ci.code[1].head == :foreigncall
 end
+
+# Test that inlining can inline _applys of builtins/_applys on SimpleVectors
+function foo_apply_apply_type_svec()
+    A = (Tuple, Float32)
+    B = Tuple{Float32, Float32}
+    Core.apply_type(A..., B.types...)
+end
+let ci = code_typed(foo_apply_apply_type_svec, Tuple{})[1].first
+    @test length(ci.code) == 1
+end
```

Comment:
can we make a better test than this? this one seems brittle under various compiler flags (notable, the code-coverage bot, although `code_typed` itself likes to mislead about this particular form of result also)—although perhaps this is really the best we can do, since we need to check for a particular optimization result?

perhaps we could at least improve the test to be more explicit: `ci.code == Any[ Expr(:return, Const(NTuple{3, Float32})) ]` (e.g. the expression that `code_typed` may or may not decide to replace the optimization result with)